### PR TITLE
Nginx idr caching config

### PR DESCRIPTION
**Pull request by manics**
Moves the IDR caching config from `roles/nginx-proxy/vars/example-omero.yml` to `idr-playbooks/group_vars/proxy-hosts.yml.

Originally the vars were stored in the role so that they could be public whilst the remaining IDR config was private, but following a refactoring nearly all of the IDR config is in public group-vars.

To test this Vagrant (`vagrant up idr-database idr-omero idr-gateway`) you will need https://github.com/openmicroscopy/infrastructure/pull/152

### CODE REVIEW

File: ansible/idr-playbooks/group_vars/proxy-hosts.yml [hunk 1]
```diff
@@ -1,11 +1,110 @@
...
+  keysize: 4000m
+  inactive: 30d
+  match:
+  - '"~web(client|gateway)/render_*/*"'
+- name: omerometadata
+  keysize: 300m
+  inactive: 30d
+  match:
+  - '"~webclient/metadata_*"'
+#- name: omeroshow
```
Comment (aleksandra-tarkowska) [inline ansible/idr-playbooks/group_vars/proxy-hosts.yml]: you can now remove this as is handled by ~webclient/api/*

Comment (manics) [inline ansible/idr-playbooks/group_vars/proxy-hosts.yml]: Done

File: ansible/idr-playbooks/group_vars/proxy-hosts.yml [hunk 2]
```diff
@@ -1,11 +1,103 @@
...
+# Cache the request path but not the host so that cache can be copied to
+# other servers
+nginx_proxy_cache_key: "$request_uri"
+
+nginx_proxy_cache_ignore_headers: '"Set-Cookie" "Vary" "Expires"'
+nginx_proxy_cache_hide_headers:
+- Set-Cookie
+
+nginx_proxy_cache_match_uri:
+#- '"~webclient/api/paths_to_object*"'
```
Comment (aleksandra-tarkowska) [inline ansible/idr-playbooks/group_vars/proxy-hosts.yml]: this can be removed as well

File: ansible/idr-playbooks/group_vars/proxy-hosts.yml [hunk 3]
```diff
@@ -1,11 +1,103 @@
...
+- '"~web(client|gateway)/(metadata|render)_*"'
+- '"~webclient/api/*"'
+- '"~static/*"'
+- '"~mapr/*"'
+
+#nginx_proxy_cache_match_arg:
```
Comment (manics) [inline ansible/idr-playbooks/group_vars/proxy-hosts.yml]: Yes, I deliberately left that line as a reminder in case we come back to it.

**Comment by manics** [comment]
I think this needs a bug fix discovered by Ola.

**Comment by manics** [comment]
On second thoughts the bug in the caching config already existed, so I think it's easier to look at this PR purely as moving variables from one file to another, otherwise there are merge conflicts since the bug fix commit comes after other new additions to file.

**Comment by aleksandra-tarkowska** [comment]
if cache is fixed https://github.com/openmicroscopy/infrastructure/pull/159/files#diff-2f8f0d61b08d12fe59795d928cfd272cR33 do we need this PR?

**Comment by manics** [comment]
#159 is basically my "merge" branch. The intention is to split it out into separate PRs for review and rebase as they're merged.

**Comment by joshmoore** [comment]
No vetoes.

**Pull request merged**